server/index.js

```diff
@@ -35,10 +35,8 @@ app.get('/api/products', (req, res, next) => {
 
 app.get('/api/products/:productId', (req, res, next) => {
   const { productId } = req.params;
-  if (!parseInt(productId, 10)) {
-    return res.status(400).json({
-      error: 'The productId must be a positive integer.'
-    });
+  if ((!parseInt(productId, 10)) || (parseInt(productId, 10) < 1)) {
+    next(new ClientError('The productId must be a positive integer', 400));
```

Comment:
Be sure to `return` here so that your middleware doesn't continue to execute.

Reply:
Fixed